### PR TITLE
Badders/develop/wip

### PR DESCRIPTION
**Pull request by djbadders**
A few fixes and improvements including:
- Fixed issues with detecting the console and whether key presses can be captured.
- Changed the start-up to keep trying to acquire configuration and/or Profit Trailer connectivity using a exponential back off to stop thrashing.
- Changed the code to use UTC times to avoid issues when daylight saving kicks in.
- Fixed checking file timestamps to use UTC
- Added better debug info on failed calls for market data
- Fixed failure to get currency exchange rates

### CODE REVIEW

File: Core/DataObjects/PTMagicData.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ public class Application
     public string TimezoneOffset { get; set; } = "+0:00";
     public string MainFiatCurrency { get; set; } = "USD";
     public string CoinMarketCapAPIKey { get; set; }
-    public string FreeCurrencyConverterAPIKey { get; set; }
+    public string FreeCurrencyConverterAPIKey { get; set; } = "sample-api-key";
```
Comment (JackTerok) [inline Core/DataObjects/PTMagicData.cs]: That should be left empty. There's no need for a sample key.

File: PTMagic/Program.cs [hunk 2]
```diff
@@ -26,7 +26,7 @@ static void Main(string[] args)
       // Keep the app running
       for (; ; )
       {
-        Thread.Sleep(100);
+        Thread.Sleep(10000);
```
Comment (JackTerok) [inline PTMagic/Program.cs]: Why change that value?

File: Core/DataObjects/PTMagicData.cs (hunk 1 above)
Comment (djbadders) [reply Core/DataObjects/PTMagicData.cs]: This is the key they use on the website to demonstrate the API and it appears to work fine, surely this is ok as a default and folks can apply for a key and set it in the config if they want / need to?

File: PTMagic/Program.cs (hunk 2 above)
Comment (djbadders) [reply PTMagic/Program.cs]: This is essentially causing a thread interrupt 10 times a second, so seemed a bit wasteful as we are just blocking the main thread from exiting.

File: Core/DataObjects/PTMagicData.cs (hunk 1 above)
Comment (JackTerok) [reply Core/DataObjects/PTMagicData.cs]: I still don't like the idea of PTM spamming their Sample API with requests.

Comment (djbadders) [reply Core/DataObjects/PTMagicData.cs]: Yeah, it is a bit questionable... just wanted to eliminate yet another obstacle to getting the software running for users.

Shall I revert it then, or encourage users to sign up?

File: PTMagic/Program.cs (hunk 2 above)
Comment (JackTerok) [reply PTMagic/Program.cs]: Yeah makes sense

File: Core/DataObjects/PTMagicData.cs (hunk 1 above)
Comment (JackTerok) [reply Core/DataObjects/PTMagicData.cs]: Yeah. That's probably for the best.

Comment (djbadders) [reply Core/DataObjects/PTMagicData.cs]: Ok I have added a commit to revert this change.

**Comment by JackTerok** [comment]
Looks good now.

**Pull request merged**